### PR TITLE
[#110512828] Add support for expiration, host-factory, and audit-send

### PR DESCRIPTION
**Pull request by apotterri**
**Needs to be merged after https://github.com/conjurinc/api-ruby/pull/59**

New support for expiration. Bring host-factory and audit-send commands in from their respective plugins.

### CODE REVIEW

File: lib/conjur/command/users.rb [hunk 1]
```diff
@@ -136,7 +136,7 @@ class Conjur::Command::Users < Conjur::Command
       c.flag [:p,:password]
 
       c.action do |global_options,options,args|
-        username, password = Conjur::Authn.read_credentials
+        username, password = Conjur::Authn.get_credentials
```
Comment (kgilpin) [inline lib/conjur/command/users.rb]: `get_credentials` is already going to prompt; what was the trouble here?

Comment (apotterri) [inline lib/conjur/command/users.rb]: If you use `read_credentials`, you can only change a user's password if his credentials are stored in his netrc. Using `get_credentials`, you can update his password however his credentials are provided (e.g. through the environment).

Comment (kgilpin) [inline lib/conjur/command/users.rb]: Do you want `get_credentials noask: true`?

Comment (apotterri) [inline lib/conjur/command/users.rb]: Wouldn't that mean that if the user tried to change his password, but wasn't logged in, he'd get an error? Is that the behavior we want?

Comment (kgilpin) [inline lib/conjur/command/users.rb]: No :-) It's good how you've done it, thanks for the chat :-)

**Pull request merged**